relay-config/src/config.rs

```diff
@@ -643,6 +700,160 @@ impl Config {
         Ok(config)
     }
 
+    /// Override configuration with values coming from other sources (e.g. env variables or
+    /// command line parameters)
+    pub fn override_config(
+        &mut self,
+        overrides: OverridableConfig,
+    ) -> Result<&mut Self, ConfigError> {
+        let relay = &mut self.values.relay;
+
+        if let Some(upstream) = overrides.upstream {
+            relay.upstream = UpstreamDescriptor::from_str(&upstream).map_err(|err| {
+                ConfigError::new_field_error(
+                    "upstream",
+                    upstream,
+                    err.context(ConfigErrorKind::InvalidOverride),
```

Comment:
The calls to `context` seem redundant. You could pull them into `ConfigError::new_field_error`. The signature would be generic over `E: Display + Send + Sync + 'static`.

Reply:
need to discuss further

Reply:
At the end the signature is:
``` rust
    fn new_field_error<E>(name: &'static str, inner: E) -> Self
    where
        E: Fail
```
Making it :
``` rust
    fn new_field_error<E>(name: &'static str, inner: E) -> Self
    where
        E: Display + Send + 'static
```
Would also work but I would loose the stack information for Failures that have stack informations ( like Errors).

Reply:
That makes sense, thanks! Let's keep `Fail` here for now and reevaluate, once we're moving to `Error`.